### ISSUE TITLE
Remix Vite plugin can't detect preamble - Vite + @remix-run/testing

### ISSUE THREAD

**Issue by depsimon** [remix-run/remix#8788]
### Reproduction

https://stackblitz.com/edit/remix-run-remix-tu6whh?file=tests%2Froutes%2F_index.test.tsx

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 832.48 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Yarn: 1.22.19 - ~/.volta/bin/yarn
    npm: 10.2.3 - ~/Library/Application Support/Herd/config/nvm/versions/node/v20.10.0/bin/npm
    pnpm: 8.14.0 - ~/.volta/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 120.1.61.104
    Chrome: 120.0.6099.234
    Edge: 120.0.2210.144
    Safari: 17.2.1
```

### Used Package Manager

npm (I'm using bun locally, but reproduced with npm on Stackblitz)

### Expected Behavior

I expect the tests to run fine when using createRemixStub with @testing-library/react in a vite app with vitest.

### Actual Behavior

Vitest throws an error `Remix Vite plugin can't detect preamble. Something is wrong.`

```
tests/routes/_index.test.tsx
Error: Remix Vite plugin can't detect preamble. Something is wrong.
 - /app/routes/_index.tsx:13:7
 - /tests/routes/_index.test.tsx:3:31
```

**Comment by pcattori** [comment]
See https://remix.run/docs/en/main/future/vite#plugin-usage-with-other-vite-based-tools-eg-vitest-storybook

**Comment by Cmoen11** [comment]
Im not sure this is actually the solution to this. I have added the `happy-dom` that was suggested, but still have this issue with vitest. 

`` 
import { installGlobals } from "@remix-run/node";
import { vitePlugin as remix } from "@remix-run/dev";
import { defineConfig } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";

installGlobals();

export default defineConfig({
	plugins: [
		remix({
			ignoredRouteFiles: ["**/.*"],
		}),
		tsconfigPaths(),
	],
	ssr: {
		noExternal: ["remix-i18next"],
	},
	test: {
		environment: "happy-dom",
	},
});
```

test

```
import { expect, test } from "vitest";
import { MyTestComponent } from "./TestComponent";
import { render, screen } from "@testing-library/react";
import { createRemixStub } from "@remix-run/testing";

test("adds 1 + 2 to equal 3", async () => {
	const Comp = createRemixStub([
		{
			path: "/",
			Component: MyTestComponent,
			loader() {
				return null;
			},
		},
	]);

	render(<Comp />);
	expect(screen.findByText("test"));
});

```
```

export const MyTestComponent = () => <p>test</p>;
```

 FAIL  app/components/TestComponent.test.tsx [ app/components/TestComponent.test.tsx ]
Error: Remix Vite plugin can't detect preamble. Something is wrong.
 ❯ app/components/TestComponent.tsx:1:38
      1| export const MyTestComponent = () => <p>test</p>;
       |                                      ^
      2| 
 ❯ app/components/TestComponent.test.tsx:3:31

**Comment by depsimon** [comment]
> EDIT: Yes it is the solutution, but it is a bit unclear from the docs that you need to conditionally load the remix plugin `!process.env.VITEST && remix(..`

I agree. I created a PR to fix the docs #8854
